src/breakupModel/simulation/Breakup.h

```diff
@@ -224,6 +238,26 @@ class Breakup {
      */
     std::array<double, 3> calculateVelocityVector(double velocity);
 
+
+    /**
+     * Returns a random number according to a specific distribution.
+     * @tparam Distribution - should be a Distribution (e. g. a uniform Distribution) which returns doubles
+     * @param distribution
+     * @return a random number
+     * @note This method is thread safe because if different threads call this they will have different random number
+     * generators (for their whole life time)
+     */
+    template<class Distribution>
+    double getRandomNumber(Distribution &distribution) {
+        if (_fixRNG.has_value()) {
+            const std::lock_guard<std::mutex> lock(_rngMutex);
+            return distribution(_fixRNG.value());
+        } else {
+            thread_local std::mt19937 rng{std::random_device{}()};
```

Comment:
just to be sure: This exists not until the end of the block but for the entire lifetime of the thread, right? Otherwise, this will  mess up your runtime because recreating random engines is expensive

Reply:
Exactly 👍